Index: src/app/feature/customer/component/customer-loan-information/customer-loan-information.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/customer/component/customer-loan-information/customer-loan-information.component.ts b/src/app/feature/customer/component/customer-loan-information/customer-loan-information.component.ts
--- a/src/app/feature/customer/component/customer-loan-information/customer-loan-information.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/customer/component/customer-loan-information/customer-loan-information.component.ts	(date 1630383470172)
@@ -455,11 +455,13 @@
         this.netTradingAssets = data;
         this.customerInfoService.saveLoanInfo(this.netTradingAssets, this.customerInfoId, TemplateName.NET_TRADING_ASSETS)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, ' Successfully saved Net Trading Assets!'));
                 this.itemNetTradingAssets.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save Successfully saved Net Trading Assets)!'));
             });
     }
@@ -503,11 +505,13 @@
         this.borrowerPortfolio = data;
         this.customerInfoService.saveLoanInfo(this.borrowerPortfolio, this.customerInfoId, TemplateName.BORROWER_PORTFOLIO)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, ' Successfully saved Borrower Portfolio!'));
                 this.loanPortfolio.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save  Borrower Portfolio!'));
             });
     }
@@ -519,11 +523,13 @@
         this.microBorrowerFinancial = data;
         this.customerInfoService.saveLoanInfo(this.microBorrowerFinancial, this.customerInfoId, TemplateName.BASEL_RISK_EXPOSURE)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, ' Successfully saved Basel Wise Risk Exposure!'));
                 this.baselRiskAccordion.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save Basel Wise Risk Exposure!'));
             });
     }
@@ -535,11 +541,13 @@
         this.borrowerPortfolio = data;
         this.customerInfoService.saveLoanInfo(this.borrowerPortfolio, this.customerInfoId, TemplateName.MICRO_BORROWER_FINANCIAL)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, ' Successfully saved Borrower Portfolio!'));
                 this.borrowerFinancialHighlight.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save  Borrower Portfolio!'));
             });
     }
@@ -599,11 +607,13 @@
         this.commentsDataResponse = data;
         this.customerInfoService.saveLoanInfo(this.commentsDataResponse, this.customerInfoId, TemplateName.COMMENTS)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, ' Successfully saved Comments!'));
                 this.commentsFromAccount.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save Comments)!'));
             });
     }
@@ -615,11 +625,13 @@
         this.securityDataResponse = data;
         this.customerInfoService.saveLoanInfo(this.securityDataResponse, this.customerInfoId, TemplateName.PREVIOUS_SECURITY)
             .subscribe(() => {
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.SUCCESS, 'Successfully saved Previous Security'));
                 this.dataFromPreviousSecurity.close();
                 this.triggerCustomerRefresh.emit(true);
             }, error => {
                 console.error(error);
+                this.overlay.hide();
                 this.toastService.show(new Alert(AlertType.ERROR, 'Unable to save Previous Security'));
             });
     }
Index: src/app/feature/loan-information-template/security/security.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/loan-information-template/security/security.component.ts b/src/app/feature/loan-information-template/security/security.component.ts
--- a/src/app/feature/loan-information-template/security/security.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/loan-information-template/security/security.component.ts	(date 1630324402808)
@@ -24,6 +24,7 @@
 import {SecurityCoverageAutoCommercial} from '../model/security-coverage-auto-commercial';
 import {Alert, AlertType} from '../../../@theme/model/Alert';
 import {ToastService} from '../../../@core/utils';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
     selector: 'app-security',
Index: src/app/feature/micro-loan/template/borrower-financial/borrower-financial.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/micro-loan/template/borrower-financial/borrower-financial.component.ts b/src/app/feature/micro-loan/template/borrower-financial/borrower-financial.component.ts
--- a/src/app/feature/micro-loan/template/borrower-financial/borrower-financial.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/micro-loan/template/borrower-financial/borrower-financial.component.ts	(date 1630383235370)
@@ -6,6 +6,7 @@
 import {ModalResponse} from '../../../../@core/utils';
 import {FinancialDeleteComponentComponent} from '../../../loan-information-template/financial/financial-delete-component/financial-delete-component.component';
 import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
   selector: 'app-borrower-financial',
@@ -23,6 +24,7 @@
   currentFormData;
 
   constructor(protected formBuilder: FormBuilder,
+              private overlay: NgxSpinnerService,
               protected modalService: NgbModal) {
   }
 
@@ -101,6 +103,7 @@
   }
 
   onSubmit() {
+    this.overlay.show();
     const currentFormData = {
       fiscalYear: this.fiscalYear,
       auditorList: this.auditorList,
Index: src/app/feature/loan-information-template/net-trading-assets/net-trading-assets.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/loan-information-template/net-trading-assets/net-trading-assets.component.ts b/src/app/feature/loan-information-template/net-trading-assets/net-trading-assets.component.ts
--- a/src/app/feature/loan-information-template/net-trading-assets/net-trading-assets.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/loan-information-template/net-trading-assets/net-trading-assets.component.ts	(date 1630382869110)
@@ -7,6 +7,7 @@
 import {ToastService} from '../../../@core/utils';
 import {Alert, AlertType} from '../../../@theme/model/Alert';
 import {CalendarType} from '../../../@core/model/calendar-type';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
     selector: 'app-net-trading-assets',
@@ -40,6 +41,7 @@
 
     constructor(protected formBuilder: FormBuilder,
                 protected fiscalYearService: FiscalYearService,
+                private overlay: NgxSpinnerService,
                 protected toastService: ToastService) {
     }
 
@@ -278,6 +280,7 @@
     }
 
     onSubmit() {
+        this.overlay.show();
         if (!ObjectUtil.isEmpty(this.netTradingAssetsData)) {
             this.netTradingAssetSubmitData = this.netTradingAssetsData;
         }
Index: src/app/feature/micro-loan/template/borrower-loan-portofolio/borrower-loan-portofolio.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/micro-loan/template/borrower-loan-portofolio/borrower-loan-portofolio.component.ts b/src/app/feature/micro-loan/template/borrower-loan-portofolio/borrower-loan-portofolio.component.ts
--- a/src/app/feature/micro-loan/template/borrower-loan-portofolio/borrower-loan-portofolio.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/micro-loan/template/borrower-loan-portofolio/borrower-loan-portofolio.component.ts	(date 1630383429422)
@@ -5,6 +5,7 @@
 import {ToastService} from '../../../../@core/utils';
 import {BorrowerPortfolio} from '../../../loan/model/borrwerportfolio';
 import {ObjectUtil} from '../../../../@core/utils/ObjectUtil';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
     selector: 'app-borrower-loan-portofolio',
@@ -23,6 +24,7 @@
 
     constructor(private formBuilder: FormBuilder,
                 private elementRef: ElementRef,
+                private overlay: NgxSpinnerService,
                 private toastService: ToastService) {
     }
 
@@ -117,8 +119,10 @@
     }
 
     submitForm() {
+        this.overlay.show();
         this.submitted = true;
         if (this.portfolioForm.invalid) {
+            this.overlay.hide();
             this.toastService.show(new Alert(AlertType.ERROR, 'All fields are mandatory!'));
             return;
         }
Index: src/app/feature/loan-information-template/comments/comments.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/loan-information-template/comments/comments.component.ts b/src/app/feature/loan-information-template/comments/comments.component.ts
--- a/src/app/feature/loan-information-template/comments/comments.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/loan-information-template/comments/comments.component.ts	(date 1630325360340)
@@ -5,6 +5,7 @@
 import {LocalStorageUtil} from '../../../@core/utils/local-storage-util';
 import {AffiliateId} from '../../../@core/utils/constants/affiliateId';
 import {ObjectUtil} from '../../../@core/utils/ObjectUtil';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
   selector: 'app-comments',
@@ -21,7 +22,8 @@
   submitted = false;
   comments;
   auditor;
-  constructor(private formBuilder: FormBuilder) { }
+  constructor(private formBuilder: FormBuilder,
+              private overlay: NgxSpinnerService) { }
 
   ngOnInit() {
     this.buildForm();
@@ -47,6 +49,7 @@
   }
 
   submitForm() {
+    this.overlay.show();
     this.submitted = true;
     if (!ObjectUtil.isEmpty(this.commentsDataResponse)) {
       this.commentsDataObject = this.commentsDataResponse;
Index: src/app/feature/loan-information-template/previous-security/previous-security.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/loan-information-template/previous-security/previous-security.component.ts b/src/app/feature/loan-information-template/previous-security/previous-security.component.ts
--- a/src/app/feature/loan-information-template/previous-security/previous-security.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/loan-information-template/previous-security/previous-security.component.ts	(date 1630325692908)
@@ -3,6 +3,7 @@
 import {FormBuilder, FormGroup} from '@angular/forms';
 import {ObjectUtil} from '../../../@core/utils/ObjectUtil';
 import {Comments} from '../../admin/modal/comments';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
     selector: 'app-previous-security',
@@ -21,7 +22,8 @@
     securityDataObject = new Comments();
     checked = false;
 
-    constructor(private formBuilder: FormBuilder) {
+    constructor(private formBuilder: FormBuilder,
+                private overlay: NgxSpinnerService) {
     }
 
     ngOnInit() {
@@ -56,6 +58,7 @@
     }
 
     submitForm() {
+        this.overlay.show();
         this.submitted = true;
         if (!ObjectUtil.isEmpty(this.securityDataResponse)) {
             this.securityDataObject = this.securityDataResponse;
Index: src/app/feature/micro-loan/template/basel-risk-exposure/basel-risk-exposure.component.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/app/feature/micro-loan/template/basel-risk-exposure/basel-risk-exposure.component.ts b/src/app/feature/micro-loan/template/basel-risk-exposure/basel-risk-exposure.component.ts
--- a/src/app/feature/micro-loan/template/basel-risk-exposure/basel-risk-exposure.component.ts	(revision cdbc81c4393af65cde4abc86e58dffa0d714f4fc)
+++ b/src/app/feature/micro-loan/template/basel-risk-exposure/basel-risk-exposure.component.ts	(date 1630383328089)
@@ -4,6 +4,7 @@
 import {ToastService} from '../../../../@core/utils';
 import {ObjectUtil} from '../../../../@core/utils/ObjectUtil';
 import {Alert, AlertType} from '../../../../@theme/model/Alert';
+import {NgxSpinnerService} from "ngx-spinner";
 
 @Component({
   selector: 'app-basel-risk-exposure',
@@ -20,6 +21,7 @@
 
   constructor(private formBuilder: FormBuilder,
               private elementRef: ElementRef,
+              private overlay: NgxSpinnerService,
               private toastService: ToastService) {
   }
 
@@ -105,9 +107,11 @@
   }
 
   public submitForm() {
+    this.overlay.show();
     this.submitted = true;
     if (this.baselRiskExposureForm.invalid) {
       this.scrollToInvalidControl();
+      this.overlay.hide();
       this.toastService.show(new Alert(AlertType.ERROR, 'Please fill all fields!'));
       return;
     }
